staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -39,6 +39,7 @@ type GarbageCollectionPolicy string
 const (
 	DeleteDependents GarbageCollectionPolicy = "DeleteDependents"
 	OrphanDependents GarbageCollectionPolicy = "OrphanDependents"
+	Unsupported      GarbageCollectionPolicy = "Unsupported"
```

Comment:
Also, it seems if someone deletes an "unsupported" resource with either "Orphan" or "foreground deletion", the object will be [deleted immediately](https://github.com/deads2k/kubernetes/blob/312fb1e1fa198f4715598feac659f5eeffd05032/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L970).

Can we add some integration tests? It's hard to enumerate all the cases in mind.